app/controllers/application_controller.rb

```diff
@@ -3,8 +3,122 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+    enable :sessions
     set :public_folder, 'public'
     set :views, 'app/views'
+    set :session_secret, "fwitter"
   end
 
+  get '/' do
+
+    erb :index
+  end
+
+  get '/signup' do
+    redirect("/tweets") if logged_in?(session)
+
+    erb :signup
+  end
+  
+  post '/signup' do
+    @user = User.new(params)
+
+    if @user.save
+      session[:id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+    
+  end
+  
+  get '/login' do
+    redirect("/tweets") if logged_in?(session)
+    erb :login
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:id] = @user.id
+    else
+      redirect "/login"
+    end
+    redirect "/tweets"
+  end
+
+  get '/logout' do
+    session.clear
+    redirect "/login"
+  end
+  
+  get '/tweets' do
+    @tweets = Tweet.all
+    @user = current_user(session)
+    if logged_in?(session)
+      erb :"tweets/index"
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets/new' do
+    redirect("/login") if !logged_in?(session)
+    erb :'tweets/new'
+  end
+
+  post '/tweets/new' do
+    redirect back if params[:content].empty?
+    @tweet = Tweet.create(params)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.